src/main.rs

```diff
@@ -142,7 +158,17 @@ impl State {
         PrivateKeyInfo::from_der(key.as_ref())?;
         Certificate::from_der(crt.as_ref())?;
 
-        Ok(State { crt, san, key })
+        let config_obj = match config {
```

Comment:
Do we really need these `_obj` suffixes?
Why not just `config`?
That would even make it possible to use short-hard in struct init below

Reply:
This differentiates between the path to the configuration file, versus the configuration object.

Reply:
What for?

Reply:
Fixed.